projects/igniteui-angular/src/lib/slider/slider.component.ts

```diff
@@ -117,6 +290,47 @@ export class IgxSliderComponent implements ControlValueAccessor, EditorProvider,
     @Input()
     public thumbLabelVisibilityDuration = 750;
 
+
+    /**
+     * Enables `labelView`, by accepting a collection of primitive values with more than one element.
+     * Each element will be equally spread over the slider and it will serve as a thumb label.
+     * Once the property is set, it will precendence over {@link maxValue}, {@link minValue}, {@link step}.
+     * This means that the manipulation for those properties won't be allowed.
+     */
+    @Input()
+    public get labels() {
+        return this._labels;
+    }
+
+    public set labels(labels: Array<number|string|boolean|null|undefined>) {
+        this._labels = labels;
+
+        this._pMax = 1;
+
+        if (this._hasViewInit) {
+            this.stepDistance = this.calculateStepDistance();
+            this.positionHandlesAndUpdateTrack();
+            this.setTickInterval(labels);
+        }
+    }
+
+    /**
+     * Returns the template context corresponding
+     * to {@link IgxThumbFromTemplateDirective} and {@link IgxThumbToTemplateDirective} templates.
+     *
+     * return {
+     *  $implicit: {@link value},
+     *  labels: {@link labels}
+     * }
+     * ```
+     */
+    public get context(): any {
+        return {
```

Comment:
Please introduce a new type/interface - [Example](https://github.com/IgniteUI/igniteui-angular/pull/4760/files#diff-bdd8572cb97955b148467d57a3fd31e4R4869). Using of Anonymous objects is not a good practice.